src/Microsoft.DotNet.Build.Tasks/GenerateEncodingTable.cs

```diff
@@ -0,0 +1,395 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateEncodingTable : Task
+    {
+        private const string CommentIndicator = "#";
+        private const char FieldDelimiter = ';';
+
+        [Required]
+        public string IANAMappings { get; set; }
+
+        [Required]
+        public string PreferedIANANames { get; set; }
+
+        [Required]
+        public string OutputDataTable { get; set; }
+
+        public string Namespace { get; set; }
+        public string ClassName { get; set; }
```

Comment:
please ignore this one. you are right there is no optional attribute. I confused this with other issue :-)